core/src/main/java/hudson/model/BooleanParameterValue.java

```diff
@@ -54,7 +54,9 @@ public BooleanParameterValue(String name, boolean value, String description) {
     @Override
     public void buildEnvVars(AbstractBuild<?,?> build, EnvVars env) {
         env.put(name,Boolean.toString(value));
-        env.put(name.toUpperCase(Locale.ENGLISH),Boolean.toString(value)); // backward compatibility pre 1.345
+        if (!env.isCaseSensitive()) {
+            env.put(name.toUpperCase(Locale.ENGLISH),Boolean.toString(value)); // backward compatibility pre 1.345
+        }
```

Comment:
I can't help feeling that you will miss things in plugins that also contribute environment.
Surely this would be better done on the actual EnvVars?